### PR TITLE
InputCommon: Implement full mouse support

### PR DESCRIPTION
**Pull request by german77**
Instead of treating the mouse as part of the window. We create a mouse device and treat it like any other controller. This allow us to easily emulate any input like analog, button, motion and touch. That way we can map it to any button from the mouse without modifying the existing structure.

This removes the need of motion-emu as it's now part of the mouse input.

TODO:
- Support camera panning with mouse
- Migrate touch input from the window to the mouse input class

### CODE REVIEW

File: src/input_common/mouse/mouse_input.h [hunk 1]
```diff
@@ -0,0 +1,98 @@
+// Copyright 2020 Yuzu Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#pragma once
```
Comment (lioncash) [inline src/input_common/mouse/mouse_input.h:5]: Missing newline after the pragma

File: src/input_common/mouse/mouse_input.h [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+     * @param y the y-coordinate of the cursor
+     */
+    void MouseMove(int x, int y);
+
+    /**
+     * Signals that a motion sensor tilt has ended.
+     */
+    void ReleaseButton(int button_);
+
+    Common::SPSCQueue<MouseStatus>& GetMouseQueue();
```
Comment (lioncash) [inline src/input_common/mouse/mouse_input.h:68]: These accessors should have a [[nodiscard]] attribute.

**Comment by Morph1984** [comment]
The fix rebase errors commit must be squashed into the previous commit for a cleaner commit history

File: src/input_common/mouse/mouse_input.h [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+namespace MouseInput {
+
+enum class MouseButton {
+    Left,
+    Weel,
+    Right,
+    Left2,
+    right2,
+    Undefined,
+};
```
Comment (Morph1984) [inline src/input_common/mouse/mouse_input.h:26]:
```suggestion
enum class MouseButton {
    Left,
    Wheel,
    Right,
    Left2,
    Right2,
    Undefined,
};
```

What is Left2 and Right2? IIRC Qt only supports the following buttons:

Left, Right, Wheel, Forward, Backward

Forward and Backward are these keys here 
image

File: src/input_common/mouse/mouse_input.cpp [hunk 4]
```diff
@@ -0,0 +1,125 @@
+// Copyright 2020 Yuzu Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
```
Comment (Morph1984) [inline src/input_common/mouse/mouse_input.cpp:3]:
```suggestion
// Copyright 2020 yuzu Emulator Project
// Licensed under GPLv2 or any later version
// Refer to the license.txt file included.
```

File: src/input_common/mouse/mouse_input.h [hunk 5]
```diff
@@ -0,0 +1,99 @@
+// Copyright 2020 Yuzu Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
```
Comment (Morph1984) [inline src/input_common/mouse/mouse_input.h:3]:
```suggestion
// Copyright 2020 yuzu Emulator Project
// Licensed under GPLv2 or any later version
// Refer to the license.txt file included.
```

File: src/input_common/mouse/mouse_poller.cpp [hunk 6]
```diff
@@ -0,0 +1,261 @@
+// Copyright 2020 Yuzu Emulator Project
+// Licensed under GPLv2 or any later version
+// Refer to the license.txt file included.
```
Comment (Morph1984) [inline src/input_common/mouse/mouse_poller.cpp:3]:
```suggestion
// Copyright 2020 yuzu Emulator Project
// Licensed under GPLv2 or any later version
// Refer to the license.txt file included.
```

File: src/input_common/mouse/mouse_poller.h [hunk 7]
```diff
@@ -0,0 +1,109 @@
+// Copyright 2020 Yuzu Emulator Project
+// Licensed under GPLv2 or any later version
+// Refer to the license.txt file included.
```
Comment (Morph1984) [inline src/input_common/mouse/mouse_poller.h:3]:
```suggestion
// Copyright 2020 yuzu Emulator Project
// Licensed under GPLv2 or any later version
// Refer to the license.txt file included.
```

**Review by bunnei — APPROVED** [review]

**Pull request merged**